Assets/Scripts/Puzzle/Gate.cs

```diff
@@ -33,19 +35,24 @@ public class Gate : SignalReceiver {
 
 	private IEnumerator ShowEvent() {
 		playerController.AllowControls = false;
-		float oldTransDur = camera.TransitionDuration;
-		float newTransDur = oldTransDur * cameraTransitionMultiplier * Vector2.Distance(transform.position, player.transform.position);
+		Time.timeScale = 0;
+		float newTransitionDuration = cameraTransitionDuration *
+		                              cameraTransitionMultiplier *
+		                              Vector2.Distance(transform.position, player.transform.position);
```

Comment:
`player` is a transform.
```suggestion
		                              Vector2.Distance(transform.position, player.position);
```